### PR TITLE
OS-8108 Support vnc password for BHYVE

### PR DESCRIPTION
**Pull request by jasonbking**

### CODE REVIEW

**Comment by sjorge** [comment]
Is there a counter part somewhere for zhyve/bhyve? I thought upstream didn't support setting passwords.

**Comment by jasonbking** [comment]
https://github.com/joyent/illumos-joyent/pull/256 -- it did but it was never wired up in the port (turned out to be easier than originally thought).

**Comment by sjorge** [comment]
LGTM, doing a build now to test.

**Comment by sjorge** [comment]
I did spend 10 minutes wondering why it was no longer listening, but then I remembred the old bug where I need to restart vmadmd after making a change to vnc_port.

```
[root@boron ~]# netstat -na | grep LISTEN
127.0.0.1.4999            *.*                 0      0 1048576      0 LISTEN
      *.111                *.*                 0      0 1048576      0 LISTEN
      *.111                *.*                 0      0 1048576      0 LISTEN
      *.22                 *.*                 0      0 1048576      0 LISTEN
127.0.0.1.9090             *.*                 0      0 1048576      0 LISTEN
127.0.0.1.8080             *.*                 0      0 1048576      0 LISTEN
127.0.0.1.25               *.*                 0      0 1048576      0 LISTEN
127.0.0.1.587              *.*                 0      0 1048576      0 LISTEN
      *.111                             *.*                              0      0 1048576      0 LISTEN
      *.22                              *.*                              0      0 1048576      0 LISTEN
::1.25                                  *.*                              0      0 1048576      0 LISTEN
[root@boron ~]# svcadm restart vmadmd
[root@boron ~]#netstat -na | grep LISTEN
127.0.0.1.4999            *.*                 0      0 1048576      0 LISTEN
      *.111                *.*                 0      0 1048576      0 LISTEN
      *.111                *.*                 0      0 1048576      0 LISTEN
      *.22                 *.*                 0      0 1048576      0 LISTEN
127.0.0.1.9090             *.*                 0      0 1048576      0 LISTEN
127.0.0.1.25               *.*                 0      0 1048576      0 LISTEN
127.0.0.1.587              *.*                 0      0 1048576      0 LISTEN
127.0.0.1.8080             *.*                 0      0 1048576      0 LISTEN
10.23.40.211.5901          *.*                 0      0 1048576      0 LISTEN
      *.111                             *.*                              0      0 1048576      0 LISTEN
      *.22                              *.*                              0      0 1048576      0 LISTEN
::1.25                                  *.*                              0      0 1048576      0 LISTEN
```

It's working fine, I get prompted for a password!

The man page does need updating though:

```
       vnc_password:

           This property allows you to set a password which will be required when
           connecting to the VNC port. IMPORTANT: this password will be visible
           from the GZ of the CN and anyone with access to the serial port in the
           guest. Set to an empty string (default) to not require a password at
           this level.

           type: string (8 chars max)
           vmtype: kvm
           listable: no
           create: yes
           update: yes
           default: <unset>
```

- vmtype should now be `HVM` instead of kvm as it wokrs on both kvm and bhyve
- there is no qmp (or what is the qemu thing called) to see the password from a guest in bhyve
- it is visible in the gz via pargs though (`30:0,fbuf,vga=off,unix=/tmp/vm.vnc,password=MYPASS`)

**Comment by sjorge** [comment]
```
       vnc_password:

           This property allows you to set a password which will be required when
           connecting to the VNC port. IMPORTANT: this password will be visible
           from the GZ of the CN. For KVM anyone with access to the serial port in the
           guest can also see the password. Set to an empty string (default) to not
           require a password at this level.

           Changing the password will require a reboot of the zone before the change
           becomes active.

           type: string (8 chars max)
           vmtype: HVM
           listable: no
           create: yes
           update: yes
           default: <unset>
```

I think the above should be fine for the new man page entry for vnc_password.

File: src/vm/node_modules/proptable.js [hunk 1]
```diff
@@ -1999,6 +1999,7 @@ exports.properties = {
         loadValueTranslator: 'utils.unbase64',
         payload: {
             allowed: {
+                'bhyve': ['create', 'receive', 'update'],
```
Comment (bahamat) [inline src/vm/node_modules/proptable.js]: The vmadm man page should also be updated to indicate that `vnc_password` is supported for both HVM types.

Comment (bahamat) [reply src/vm/node_modules/proptable.js]: Also copyright in this file?

Comment (sjorge) [reply src/vm/node_modules/proptable.js]: @bahamat yeah I noticed it too, see https://github.com/joyent/smartos-live/pull/900#issuecomment-582820863 as there are some additional things to mention/update. 

Anything I've missed?

Comment (bahamat) [reply src/vm/node_modules/proptable.js]: Ah, yeah. I missed that whole conversation. I was trying to hurriedly look over the change just before another meeting and didn't look at the comments. I think that text for the man page change looks good.

File: src/vm/man/vmadm.1m.md [hunk 2]
```diff
@@ -2116,12 +2116,16 @@ tab-complete UUIDs rather than having to type them out for every command.
...
-        from the GZ of the CN and anyone with access to the serial port in the
-        guest. Set to an empty string (default) to not require a password at
-        this level.
+        from the GZ of the CN. For KVM anyone with access to the serial port in
+	the guest can also see the password. Set to an empty string (default) to
+	not require a password at this level.
+
+	Changing the password will require a reboot of the zone before the
+	change becomes active. Reboots from inside the guest will not make
+	the changed password active.
```
Comment (bahamat) [inline src/vm/man/vmadm.1m.md]: Should have consistent whitespace.

**Pull request merged**